File: src/UI/Mobile/MauiApp/Services/AuditService.cs
using EnterpriseTemplate.MauiApp.Services.Base;
using ProjectTemplate.Shared.Models;

namespace EnterpriseTemplate.MauiApp.Services;

public interface IAuditService
{
    Task<PagedResponse<AuditLogDto>?> GetAuditLogsAsync(string entityType = null, string eventType = null, int page = 1, int pageSize = 10, CancellationToken cancellationToken = default);
}

public class AuditService : BaseService, IAuditService
{
    public AuditService(HttpClient http) : base(http) { }

    public Task<PagedResponse<AuditLogDto>?> GetAuditLogsAsync(string entityType = null, string eventType = null, int page = 1, int pageSize = 10, CancellationToken cancellationToken = default)
    {
        var url = $"api/v1/audit?page={page}&pageSize={pageSize}";
        if (!string.IsNullOrEmpty(entityType)) url += $"&entityType={entityType}";
        if (!string.IsNullOrEmpty(eventType)) url += $"&eventType={eventType}";
        
        return GetAsync<PagedResponse<AuditLogDto>>(url, cancellationToken);
    }
}

File: src/UI/Web/Angular/src/app/core/services/audit.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { PagedResponse, DomainEvent } from '../../shared/models';

@Injectable({ providedIn: 'root' })
export class AuditService extends BaseService<DomainEvent> {
  protected override entityPath = 'api/v1/audit';

  getAuditLogs(page = 1, pageSize = 10, entityType?: string, eventType?: string, userId?: string, from?: Date, to?: Date) {
    let url = `${this.fullUrl}?page=${page}&pageSize=${pageSize}`;
    if (entityType) url += `&entityType=${entityType}`;
    if (eventType) url += `&eventType=${encodeURIComponent(eventType)}`;
    if (userId) url += `&userId=${encodeURIComponent(userId)}`;
    if (from) url += `&startDate=${from.toISOString()}`;
    if (to) url += `&endDate=${to.toISOString()}`;
    return this.http.get<PagedResponse<DomainEvent>>(url);
  }
}
